starwars/src/App.js

```diff
@@ -1,19 +1,72 @@
-import React from 'react';
-import './App.css';
+import React, { useState, useEffect } from "react";
+import Persons from "./components/Persons/Persons";
+import axios from "axios";
+import PaginationButtons from "./components/PaginationButtons/PaginationButtons";
+import PageLoader from "./components/Loader/Loader";
+import "./App.css";
 
 const App = () => {
   // Try to think through what state you'll need for this app before starting. Then build out
   // the state properties here.
 
-  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
+  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a
   // side effect in a component, you want to think about which state and/or props it should
   // sync up with, if any.
 
+  const [people, setPeople] = useState([]);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [disablePrev, setdisablePrev] = useState(true);
+  const [disableNext, setdisableNext] = useState(false);
+  const noOfPeoplePerPage = 10;
+  let pagesCount;
+
+  useEffect(
+    () => {
+      axios
+        .get(`https://swapi.co/api/people/?page=${pageNumber}`)
+        .then(response => {
+          const people = response.data;
+          setPeople(people.results);
+          setdisablePrev(true);
+          setdisableNext(false);
+          pagesCount = people.count;
+          const maxPages = Math.ceil(pagesCount / noOfPeoplePerPage)
+          // console.log("pageNumber = "+pageNumber +"Max = "+maxPages);
+          if (pageNumber > 1) setdisablePrev(false);
+          if (pageNumber >= Math.ceil(pagesCount / noOfPeoplePerPage)) setdisableNext(true);
+          // console.log(people.results.length);
+        })
+        .catch(err => {
+          console.log(err);
```

Comment:
Take care of the sad path next time, make sure user get to see if something goes wrong while fetching data.